scapy/layers/bluetooth.py

```diff
@@ -477,7 +580,10 @@ class HCI_LE_Meta_Advertising_Report(Packet):
                     ByteEnumField("atype", 0, {0:"public", 1:"random"}),
                     LEMACField("addr", None),
                     FieldLenField("len", None, length_of="data", fmt="B"),
-                    StrLenField("data", "", length_from=lambda pkt:pkt.len), ]
+                    PacketListField("data", [], EIR_Hdr,
+                                    length_from=lambda pkt:pkt.len),
+                    SignedCharField("rssi", 0)]
```

Comment:
Here, the field `len` contains the length of the rest of the packet except for the last byte, while the previous code was counting the length of the rest of the packet (including the last byte).

Is the existing code wrong?